presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java

```diff
@@ -42,33 +43,66 @@
 import static com.facebook.presto.spi.StandardErrorCode.COMPILER_ERROR;
 import static com.facebook.presto.sql.gen.BytecodeUtils.invoke;
 import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
 
 public class ExpressionCompiler
 {
     private final Metadata metadata;
 
-    private final LoadingCache<CacheKey, Class<? extends PageProcessor>> pageProcessors = CacheBuilder.newBuilder().maximumSize(1000).build(
-            new CacheLoader<CacheKey, Class<? extends PageProcessor>>()
+    private final LoadingCache<ProcessorCacheKey, Class<? extends PageProcessor>> pageProcessors = CacheBuilder.newBuilder().maximumSize(1000).build(
```

Comment:
Looking at the changes in this file.  The only method reused from the existing code is `generateToString` (and that reuse isn't necessary).  So, instead of adding more stuff to this class, move all of this new stuff to `JoinFilterFunctionCompiler` and make it injectable, so it can be bound directly into `LocalExecutionPlanner`.

You can replace the call to `generateToString` with this:
```java
        classDefinition.declareMethod(a(PUBLIC), "toString", type(String.class))
                .getBody()
                .append(constantString(toStringHelper(classDefinition.getType().getJavaClassName())
                        .add("filter", filter)
                        .add("leftBlocksSize", leftBlocksSize)
                        .toString()).ret());